### DISCUSSION TITLE
Eloquent: select + with , no work ?

### DISCUSSION THREAD

**Discussion by Rusbelito** [laravel/framework#53958] [Q&A]
In my learning path I understood that I should bring only the columns that I am going to use and not all the columns and this would allow me to have fast and efficient queries, but when trying to do a select and at the same time bring data from a relation it responds with the null relation:

example:

 `$users = User::select('id','name')->with('post')->get();` or  `$users = User::select(['id','name'])->with('post')->get();`

it doesn't work for me, I use `dd($users)` and I verify that in relations I get null 

```
#relations: array:1 [▼
"post" => null
]
```

if I remove "select" it correctly brings the relation data, am I doing something wrong?

**Comment by macropay-solutions** [comment]
You must select also the column that links the User to the relation.

Update.
We tested on our demo page
http://89.40.19.34/laravel-10/laravel-lumen-crud-wizard#operations
 with text area query string
`distincts[0]=client_id&withRelations[0]=client&limit=1&page=1`
client is filled

This is in L 10 and L9

Your situation is this

`distincts[0]=id&withRelations[0]=client&limit=1&page=1`

**Reply by Rusbelito** [reply]
> Debe seleccionar también la columna que vincula al usuario a la relación.
> 
> Actualizar.Lo probamos en nuestra página de demostración http://89.40.19.34/laravel-10/laravel-lumen-crud-wizard#operationscon cadena de consulta de área de texto `distincts[0]=client_id&withRelations[0]=client&limit=1&page=1`El cliente está lleno
> 
> Esto está en L 10 y L9
> 
> Su situación es la siguiente
> 
> `distincts[0]=id&withRelations[0]=client&limit=1&page=1`

Sure, I don't understand the demo page but now I understand that I need to include the relationship column, thanks <3

**Comment by Rusbelito** [comment]
Sure, I don't understand the demo page but now I understand that I need to include the relationship column, thanks <3

**Reply by macropay-solutions** [reply]
![image](https://github.com/user-attachments/assets/a8723d69-7dec-4ab0-86c0-43677cf04538)

vs client null when selecting just id

![image](https://github.com/user-attachments/assets/f851b0bd-9183-4f43-9ead-4d6dae111697)

Look at the top white textarea from distincts[0]=client_id vs distincts[0]=id
